tests/test_le_lambda.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import
+import gzip
+import os
+import unittest
+from StringIO import StringIO
+
+import mock
+
+import le_lambda
+
+os.environ.update({
+    'region': 'unknown-region',
+    'token': 'fake-token',
+    'AWS_ACCESS_KEY_ID': '-',
+    'AWS_SECRET_ACCESS_KEY': '-',
+    'AWS_SESSION_TOKEN': '-'
+})
+
+# Patch global module-level variables
+le_lambda.TOKEN = os.environ['token']
+le_lambda.REGION = os.environ['region']
+
+
+class TestLambdaHandler(unittest.TestCase):
+    @mock.patch('le_lambda.s3')
+    @mock.patch('le_lambda.create_socket', return_value=mock.MagicMock())
+    @mock.patch('le_lambda.validate_uuid', return_value=True)
+    def test_invocation(self, validate_uuid, create_socket, s3):
+        _ = validate_uuid
+        mock_socket = create_socket.return_value
+        mock_socket.sendall = mock.MagicMock()
+        s3_contents = StringIO()
+        with gzip.GzipFile(fileobj=s3_contents, mode="w") as f:
+            f.write(
+                'https 2017-09-12T18:49:38.910213Z app/'
+                '22ff55229966cc00aa9911eeaa226677/9999cceedd225588 '
+                '254.254.254.5:8750 10.0.3.100:32831 0.001 0.002 0.000 '
```

Comment:
There is one corner case here regarding the IP and port parsing:
a. the field that contains the target ip and port may contain a `-`, which will break the parsing as it is implemented in le_lambda.py right now
b. IPv6 addresses both for clients and targets cannot get parsed correctly based on the current parsing implementation

Reply:
`-` is handled now for ALB logs and test case has been added.